### DISCUSSION TITLE
Resample in one dimension only?

### DISCUSSION THREAD

**Discussion by scottassen** [Project-MONAI/MONAI#2422] [Q&A]
Is there a way to resample an image in one pixdim/dimension only?

For example, in the [spleen segmentation](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d.ipynb) dataset, the original image shapes range from (512, 512, 31) to (512, 512, 168), with axial (z) voxel dimensions of 1.5 mm to 8 mm. Is there a way to resample these images to all have the same axial (z) voxel dimensions, while leaving the x and y dimensions the same (so that the final image shapes are still all (512, 512, z))?

The Spacing transform doesn't let you keep original pixdims for some dimensions, and Resize would make all images exactly the same dimensions.

Example desired input:
```
1. image shape: torch.Size([1, 512, 512, 60])
   pixdim: [1.       0.751953 0.751953 5.       0.       0.       0.       0.      ]
2. image shape: torch.Size([1, 512, 512, 31])
   pixdim: [1.       0.785156 0.785156 7.5      0.       0.       0.       0.      ]
```

Example desired output:
```
1. image shape: torch.Size([1, 512, 512, 60])
   pixdim: [1.       0.751953 0.751953 5.       0.       0.       0.       0.      ]
2. image shape: torch.Size([1, 512, 512, 47])
   pixdim: [1.       0.785156 0.785156 5      0.       0.       0.       0.      ]
```

**Comment by Nic-Ma** [comment]
Hi @scottassen ,

Thanks for your feature request, yes, we may consider to support `pixdim=(-1, -1, X)` for your case.
In your particular example, I think you can use `Spacingd` + `Resized` transforms.
Just set `spatial_size=[512, 512, -1]` in `Resized` to keep the last dim after `Spacingd`.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/spatial/array.py#L330

Thanks.

**Reply by Nic-Ma** [reply]
I submitted PR to support this feature request: https://github.com/Project-MONAI/MONAI/pull/2423.

Thanks.

**Reply by scottassen** [reply]
Thank you! And I appreciate the tip to use Spacingd and Resized in order.
